src/features/contracts/contracts-directory.ts

```diff
@@ -125,9 +125,15 @@ const prepareContractDirectory = async (repo: RepositoryInfo) => {
 };
 
 const getRequestOptions = (repo: RepositoryInfo) => {
-	const auth = repo.token
-		? `Basic ${Buffer.from(repo.token).toString('base64')}`
-		: '';
+	const auth =
+		repo.token == null
+			? ''
+			: // legacy `username:token` Basic authentication.
+				// TODO: Consider dropping in the next major
+				/^\w+:\w+$/.test(repo.token)
```

Comment:
I am slightly concerned with this match e.g. it won't match the `-` characters and several others. Do we have the correct regex to match our db format for usernames in the db?